From 0701f84a57156f8f584bb4fe928811a26ceefcf0 Mon Sep 17 00:00:00 2001
From: Jonathan Thomas <echidnaman@kubuntu.org>
Date: Wed, 19 Jan 2011 10:21:53 -0500
Subject: [PATCH 05/20] Bring the auth dialog to the front when it is shown.
 Otherwise it pops up behind the app requesting it and
 the user never sees it.

---
 policykitlistener.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/policykitlistener.cpp b/policykitlistener.cpp
index 68185bc..bb95f1d 100644
--- a/policykitlistener.cpp
+++ b/policykitlistener.cpp
@@ -22,6 +22,7 @@
 #include "AuthDialog.h"
 
 #include <KDebug>
+#include <KWindowSystem>
 
 #include <PolkitQt1/Agent/Listener>
 #include <PolkitQt1/Agent/Session>
@@ -100,6 +101,7 @@ void PolicyKitListener::initiateAuthentication(const QString &actionId,
     kDebug() << "WinId of the dialog is " << m_dialog.data()->winId() << m_dialog.data()->effectiveWinId();
     m_dialog.data()->setOptions();
     m_dialog.data()->show();
+    KWindowSystem::forceActiveWindow(m_dialog.data()->winId());
     kDebug() << "WinId of the shown dialog is " << m_dialog.data()->winId() << m_dialog.data()->effectiveWinId();
 
     m_numTries = 0;
-- 
1.7.7.4

